pkg/api/v1beta1/types.go

```diff
@@ -1561,3 +1564,100 @@ type GlusterfsVolumeSource struct {
 	// the Glusterfs volume to be mounted with read-only permissions
 	ReadOnly bool `json:"readOnly,omitempty" description:"Glusterfs volume to be mounted with read-only permissions"`
 }
+
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
```

Comment:
We have a general concern about changing v1beta3 for v1.  Privileged is more difficult to move out of the Pod.  It may be that we simply add these individual elements to the Container directly using an Embedded object so they don't change.  For example:

```
  container {
    privileged: true,
    capabilities: ['add-network'],
    seLinuxOptions: {...},
  }
```

and in Go 

```
type Container struct {
  ...
  SecurityContext `json:",inline"`
}
```
and remove Capabilities and Privileged.  That would mean for v1beta3/v1 we would not have a nested security context field, just an embedded SecurityContext.  Which accomplishes the point of the struct (group them together).  We can even throw in HostNetwork.